sapi/cli/php_cli_server.c

```diff
@@ -1267,6 +1267,9 @@ static void php_cli_server_logf(int type, const char *format, ...) /* {{{ */
 
 static php_socket_t php_network_listen_socket(const char *host, int *port, int socktype, int *af, socklen_t *socklen, zend_string **errstr) /* {{{ */
 {
+#ifdef __wasi__
+  return -1;
+#else
```

Comment:
you should rather disable the whole cli server as there's no point to have it without fork.